crates/core/Cargo.toml

```diff
@@ -63,8 +63,9 @@ tokio = { workspace = true }
 # datafusion
 datafusion = { workspace = true, optional = true }
 datafusion-expr = { workspace = true, optional = true }
-datafusion-common = { workspace = true, optional = true }
+datafusion-common = { workspace = true }
```

Comment:
hudi-core should be optionally dependent on datafusion - only when datafusion feature is enabled. let's keep this optional

Reply:
It now relies on the `ScalarValue` of DataFusion.